### PR TITLE
Pagination added with unit test...

### PR DESCRIPTION
**Pull request by muhammad-h264ussain**

### CODE REVIEW

File: tests/test_get_vin_history.py [hunk 1]
```diff
@@ -81,3 +81,22 @@ def test_vin_non_validate(self):
...
+        
+        token = os.getenv('API_TOKEN')
+        client = MarketAPI(token)
+
+        vins = {"1FTEW1EF1FFA67753":6,}
+
+        for vin,page_limit in vins.items():
+            for count in range(1,page_limit+1):
+                time.sleep(3)
+                api_response = client.get_vin_history(vin=vin, page=count)
```
Comment (isaacm) [inline tests/test_get_vin_history.py:105]: `page` keyword arg is not supported by the api client.

File: tests/test_get_vin_history.py [hunk 2]
```diff
@@ -81,3 +81,22 @@ def test_vin_non_validate(self):
...
+    def test_get_vin_history_with_pagination(self):
+        
+        token = os.getenv('API_TOKEN')
+        client = MarketAPI(token)
+
+        vins = {"1FTEW1EF1FFA67753":6,}
+
+        for vin,page_limit in vins.items():
+            for count in range(1,page_limit+1):
+                time.sleep(3)
```
Comment (tonytan4ever) [inline tests/test_get_vin_history.py]: Is the `time.sleep` added here for ratelimit ?
If it is,  once you tested and it works with the ACTUAL api token,  please mock out the response and remove the `time.sleep` here

File: market_api/market_api.py [hunk 3]
```diff
@@ -29,11 +29,13 @@ def get_vin_history(self, vin):
...
         response = requests.request("GET",
                                     request_url,
                                     headers=self.headers,
-                                    params=self.get_querystring())
-
+                                    params=req_query)
+        
         schema = GetVinHistoryOutputSchema()
         result = []
         for history in response.json():
```
Comment (tonytan4ever) [inline market_api/market_api.py]: Does the response have the current_number of page and total number of count/page information in ?
If there is please add those in the response.  
That's why I was indicating to ask a pagination response class.

File: tests/test_get_vin_history.py (hunk 1 above)
Comment (muhammad-h264ussain) [reply tests/test_get_vin_history.py:105]: I use this as a reference for page keyword: 
image

https://apidocs.marketcheck.com/?version=latest

File: market_api/market_api.py (hunk 3 above)
Comment (muhammad-h264ussain) [reply market_api/market_api.py]: At the time pagination response class not added... I have check marketapi docs and response not contain the total number of results & page number therefor the its not possible to guess the count pages information....

Comment (muhammad-h264ussain) [reply market_api/market_api.py]: pagination class can be apply if we can get all records once... 
What if i make for loop and make multiple request for marketsearch-api and fetch all data every time and then apply pagination class is it work for your.

In current i use keyword arg and add it in header of marketsearch-api request... That's according to marketsearch-api documentation.

**Comment by isaacm** [comment]
Merged manually.

**Pull request closed without merging**